aio/aio-proxy/aio_proxy/search/search_functions.py

```diff
@@ -13,22 +18,121 @@ def search_text(index, offset: int, page_size: int, **params):
     if is_siren(query_terms):
         query_terms_clean = query_terms.replace(" ", "")
         s = filter_by_siren(s, query_terms_clean)
-        return sort_and_execute_search(search=s, offset=offset, page_size=page_size)
+        return sort_and_execute_search(
+            search=s, offset=offset, page_size=page_size, is_search_fields=False
+        )
 
     # Use filters to reduce search results
     s = filter_search(
         s,
         filters_to_ignore=[
             "terms",
-            "min_date_naiss_dirigeant",
-            "max_date_naiss_dirigeant",
+            "convention_collective_renseignee",
+            "est_finess",
+            "est_uai",
+            "est_collectivite_territoriale",
+            "est_entrepreneur_spectacle",
+            "est_rge",
+            "id_convention_collective",
+            "id_uai",
+            "id_finess",
+            "id_rge",
             "nom_dirigeant",
             "prenoms_dirigeant",
+            "min_date_naiss_dirigeant",
+            "max_date_naiss_dirigeant",
+            "nom_elu",
+            "prenoms_elu",
+            "min_date_naiss_elu",
+            "max_date_naiss_elu",
+            "nom_personne",
+            "prenoms_personne",
+            "min_date_naiss_personne",
+            "max_date_naiss_personne",
+        ],
+        **params,
+    )
+    s = filter_search_is_exist(
+        s,
+        filters_to_process=[
+            "convention_collective_renseignee",
+            "est_finess",
+            "est_uai",
+            "est_collectivite_territoriale",
+            "est_entrepreneur_spectacle",
+            "est_rge",
+        ],
+        **params,
+    )
+    s = filter_search_match_array(
+        s,
+        filters_to_process=[
+            "id_convention_collective",
+            "id_uai",
+            "id_finess",
+            "id_rge",
         ],
         **params,
     )
     # Search dirigeants
-    s = search_dirigeant(s, **params)
+    s = search_person(
+        s,
+        "nom_dirigeant",
+        "prenoms_dirigeant",
+        "min_date_naiss_dirigeant",
+        "max_date_naiss_dirigeant",
+        [
+            {
+                "nested_object": "dirigeants_pp",
+                "match_nom": "nom",
+                "match_prenom": "prenoms",
+                "match_date": "date_naissance",
+            },
+        ],
+        **params,
+    )
+    # Search élus
+    s = search_person(
+        s,
+        "nom_elu",
+        "prenoms_elu",
+        "min_date_naiss_elu",
+        "max_date_naiss_elu",
+        [
+            {
+                "nested_object": "colter_elus",
+                "match_nom": "nom",
+                "match_prenom": "prenom",
+                "match_date": "date_naissance",
+            },
+        ],
+        **params,
+    )
+
+    # Search élus and dirigeants together
+    s = search_person(
```

Comment:
I think this logic is a little too complicated, I had a hard time understanding it all.
I think we could possibly find a way to make it more simple 🤔